src/diffusers/schedulers/scheduling_ddpm.py

```diff
@@ -270,11 +290,21 @@ def step(
             predicted_variance = None
 
         # 1. compute alphas, betas
+        if prev_timestep is None:
+            prev_timestep = t - 1
+
         alpha_prod_t = self.alphas_cumprod[t]
-        alpha_prod_t_prev = self.alphas_cumprod[t - 1] if t > 0 else self.one
+        alpha_prod_t_prev = self.alphas_cumprod[prev_timestep] if prev_timestep >= 0 else self.one
         beta_prod_t = 1 - alpha_prod_t
         beta_prod_t_prev = 1 - alpha_prod_t_prev
 
+        if prev_timestep == t - 1:
```

Comment:
Interesting, let's see whether it might be simpler to just have a new scheduler here

Reply:
resolving and using https://github.com/huggingface/diffusers/pull/1428#discussion_r1051388218 as tracking conversation